pkg/agent/openflow/pipeline.go

```diff
@@ -1777,21 +1775,27 @@ func (c *client) serviceEndpointGroup(groupID binding.GroupIDType, withSessionAf
 	return group
 }
 
-// decTTLFlows decrements TTL by one for the packets from a local Pod to a remote Pod. Antrea doesn't
-// decrement TTL for the packets which enter OVS pipeline from antrea-gw0 on the source Node.
-// TTL is also decremented on the destination Node in L3Routing Table when the packet is received.
-// See the functions of "l3FlowsToPod", "l3ToPodFlow", and "l3ToGWFlow".
+// decTTLFlows decrements TTL by one for the packets forwarded across Nodes.
+// The TTL decrement should be skipped for the packets which enter OVS pipeline
+// from the gateway interface, as the host IP stack should have decremented the
+// TTL already for such packets.
 func (c *client) decTTLFlows(category cookie.Category) []binding.Flow {
 	var flows []binding.Flow
+	decTTLTable := c.pipeline[l3DecTTLTable]
 	for _, proto := range c.ipProtocols {
-		flows = append(flows, c.pipeline[l3DecTTLTable].BuildFlow(priorityNormal).MatchPriority(priorityNormal).
-			MatchProtocol(proto).
-			MatchRegRange(int(marksReg), markTrafficFromLocal, binding.Range{0, 15}).
-			Action().DecTTL().
-			// Bypass l2ForwardingCalcTable and tables for ingress rules (which won't
-			// apply to packets to remote Nodes).
-			Action().GotoTable(conntrackCommitTable).
-			Done())
+		flows = append(flows,
+			// Skip packets from the gateway interface.
+			decTTLTable.BuildFlow(priorityHigh).MatchPriority(priorityNormal).
```

Comment:
@wenyingd : I assume kube-proxy packets (even they come from tunnel/remote Node) should bypass DecTTL(). What you think?